cs/client.py

```diff
@@ -205,26 +218,24 @@ def _sign(self, data):
         return base64.b64encode(digest).decode('utf-8').strip()
 
 
+def read_env_vars():
+    env_conf = {}
+    for key, default_value in ALLOWED_CONFIGS.items():
+        env_key = 'CLOUDSTACK_{0}'.format(key.upper())
+        env_conf[key] = os.environ.get(env_key, default_value)
+    return env_conf
+
+
 def read_config(ini_group=None):
+    config = {}
     if not ini_group:
         ini_group = os.environ.get('CLOUDSTACK_REGION', 'cloudstack')
-    # Try env vars first
-    os.environ.setdefault('CLOUDSTACK_METHOD', 'get')
-    os.environ.setdefault('CLOUDSTACK_TIMEOUT', '10')
-    keys = ['endpoint', 'key', 'secret', 'method', 'timeout']
-    env_conf = {}
-    for key in keys:
-        if 'CLOUDSTACK_{0}'.format(key.upper()) not in os.environ:
-            break
```

Comment:
this `break`/`else` was preventing from reading the `cloudstack ini` if enough information was provided.